doc/devguide/extending/app-layer/transactions.rst

```diff
@@ -0,0 +1,310 @@
+************
+Transactions
+************
+
+.. contents:: Table of Contents
+
+General Concepts
+================
+
+For Suricata, transactions are abstractions that help with detecting and logging. An example of complete transaction is
+a pair of messages Request (from frontend to backend) and Response (from backend to frontend) in HTTP.
```

Comment:
I would not mention frontend and backend, but rather client and server

Reply:
> Looks overall good to me.
> It should be read by the next person who knows nothing yet about transactions to see if it is clear ;-)

Ok, I can run it by some folks, but not sure if there's a better way. Would it make sense to post it in our forum and ask for feedback, or is that too much?

Reply:
> I would not mention frontend and backend, but rather client and server

Ok. May I ask why?

Reply:
> Ok, I can run it by some folks, but not sure if there's a better way. 

No, just something to think about next time a newcomer arrives

> I would not mention frontend and backend, but rather client and server

For me client/server, as in TCP, is more self-explaining
frontend and backend are rather all parts of the server, the frontend is just the interface, and the backend is where the database, the business logic is. That may come from me being French and using words such as front-office and back-office...

Reply:
I see. Thanks for taking the time to answer. I think I got used to frontend/backend because of the Postgresql documentation. I'll make another PR incorporating your feedback regarding this and the other comments.